server/aesdsocket.c

```diff
@@ -110,244 +133,421 @@ int main(int argc, char **argv)
+        goto error_0;
+    }
+
+    /* If daemon mode is True, create daemon*/
     if(d_mode == TRUE) {
         rc = daemon(0,0);
         if (rc < 0) {
             syslog(LOG_ERR, "aesdsocket: daemon creation error %s", strerror(errno));
-            goto  return_func;
+            goto  error_0;
         }
     }
+    syslog(LOG_INFO, "aesdsocket:Starting aesdsocket_server in daemon = %s", 
+        ((d_mode ==1)? "TRUE" : "FALSE")); 
+
 
     fp =  fopen("/var/tmp/aesdsocketdata", "a+");
     if (fp == NULL) {
         syslog(LOG_ERR, "aesdsocket: File open Error %s", strerror(errno));
-        goto  return_func;
+        goto  error_0;
     }
 
     file_close = TRUE;
-
-    syslog(LOG_INFO, "aesdsocket:Starting aesdsocket_server in daemon = %s", 
-        ((d_mode ==1)? "TRUE" : "FALSE")); 
+    
+    if(asesd_soc_timer_init()) {
+        goto error_1;
+    }
 
     memset(&signal_action, 0, sizeof(signal_action));
     signal_action.sa_handler = aesdsoc_sighandler;
     rc = sigaction(SIGINT,  &signal_action, 0);
     if (rc < 0 ) {
         syslog(LOG_ERR, "aesdsocket: Sigaction failed for SIGINT %s", strerror(errno));
-        goto error_1;
+        goto error_2;
     }
     rc = sigaction(SIGTERM, &signal_action, 0);
     if (rc < 0 ) {
         syslog(LOG_ERR, "aesdsocket: Sigaction failed for SIGTERM %s", strerror(errno));
-        goto error_1;
+        goto error_2;
     }
-
-    soc_server = socket(AF_INET, SOCK_STREAM, 0);
-    if (soc_server < 0) {
-        rc = -1;
-        syslog(LOG_ERR, "aesdsocket: Socket creation failed %s", strerror(errno));
-        goto error_1;
-    }
-
-    rc = setsockopt(soc_server, SOL_SOCKET, SO_REUSEPORT, &cmd_option, sizeof(cmd_option));
-    if (rc < 0) {
-        syslog(LOG_ERR, "aesdsocket: API setsockopt failed %s", strerror(errno));
-        goto exit_shutdown;
-    }
-
-    aesdsoc_addr.sin_family = AF_INET;
-    aesdsoc_addr.sin_addr.s_addr = INADDR_ANY;
-    aesdsoc_addr.sin_port = htons(SOCKET_PORT);
-
-    rc = bind(soc_server, (struct sockaddr*)&aesdsoc_addr, aesdsoc_addr_len);
-    if (rc < 0) {
-        syslog(LOG_ERR, "aesdsocket: API bind failure %s", strerror(errno));
-        goto exit_shutdown;
-    }
-
-    while( exit_aesd_soc == FALSE) {
-        rc = listen(soc_server, MAX_SERVER_CONNECTION);
-        if (rc < 0) {
-            syslog(LOG_ERR, "aesdsocket: API listen failed %s", strerror(errno));
-            goto exit_shutdown;
-        }
-
+    
+    syslog(LOG_INFO,"**** AESDSOCKET application: Staring multi-thread mode ****");
+    while( exit_aesd_soc == FALSE) {            
+        syslog(LOG_INFO,"**** AESDSOCKET application: accept ****");
         soc_client = accept(soc_server, (struct sockaddr*)&aesdsoc_addr, 
             (socklen_t*)&aesdsoc_addr_len);
         if (soc_client < 0 ) {
             rc = -1;
             syslog(LOG_ERR, "aesdsocket: API accept failed %s", strerror(errno));
-            goto exit_shutdown;
+            goto error_2;
```

Comment:
@sujoyray-ucolorado I was expecting to see handling for E_WOULDBLOCK here.  I would expect non blocking accept would return a negative value and `E_WOULDBLOCK` on the second call after the `fcntl` below.